ec/ecresource/deploymentresource/observability/v1/schema.go

```diff
@@ -18,47 +18,31 @@
 package v1
 
 import (
-	"github.com/elastic/terraform-provider-ec/ec/internal/planmodifier"
-	"github.com/hashicorp/terraform-plugin-framework-validators/listvalidator"
-	"github.com/hashicorp/terraform-plugin-framework/resource"
-	"github.com/hashicorp/terraform-plugin-framework/tfsdk"
-	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
 )
 
-func ObservabilitySchema() tfsdk.Attribute {
-	return tfsdk.Attribute{
+func ObservabilitySchema() schema.Attribute {
+	return schema.ListNestedAttribute{
 		Description: "Optional observability settings. Ship logs and metrics to a dedicated deployment.",
 		Optional:    true,
-		Validators:  []tfsdk.AttributeValidator{listvalidator.SizeAtMost(1)},
-		Attributes: tfsdk.ListNestedAttributes(map[string]tfsdk.Attribute{
-			"deployment_id": {
-				Type:     types.StringType,
-				Required: true,
-			},
-			"ref_id": {
-				Type:     types.StringType,
-				Computed: true,
-				Optional: true,
-				PlanModifiers: tfsdk.AttributePlanModifiers{
-					resource.UseStateForUnknown(),
+		NestedObject: schema.NestedAttributeObject{
+			Attributes: map[string]schema.Attribute{
+				"deployment_id": schema.StringAttribute{
+					Required: true,
 				},
-			},
-			"logs": {
-				Type:     types.BoolType,
-				Optional: true,
-				Computed: true,
-				PlanModifiers: []tfsdk.AttributePlanModifier{
-					planmodifier.DefaultValue(types.Bool{Value: true}),
+				"ref_id": schema.StringAttribute{
+					Computed: true,
+					Optional: true,
 				},
-			},
-			"metrics": {
-				Type:     types.BoolType,
-				Optional: true,
-				Computed: true,
-				PlanModifiers: []tfsdk.AttributePlanModifier{
-					planmodifier.DefaultValue(types.Bool{Value: true}),
+				"logs": schema.BoolAttribute{
+					Optional: true,
```

Comment:
Why do we drop the default values here but not in the v2 schema?

Reply:
I think it doesn't make sense to have plan modifiers and validators in v1 - the only need of v1 is to support state upgrade (once we decide to implement it) so currently v1 isn't used at all. Also _AFAIK_ both plan modifiers and validators aren't invoked during state upgrade.